### PR TITLE
Add SchedulerTargetFmaxMhzINTEL execution mode

### PR DESCRIPTION
**Pull request by MrSidims**
Indicates the target clock frequency (Fmax) for the kernel, in MHz.
Only valid with the Kernel Execution Model.

Spec:
https://github.com/KhronosGroup/SPIRV-Registry/blob/master/extensions/INTEL/SPV_INTEL_kernel_attributes.asciidoc

Signed-off-by: Dmitry Sidorov <dmitry.sidorov@intel.com>

### CODE REVIEW

**Review by mlychkov — APPROVED** [review]

File: test/transcoding/IntelFPGAFunctionAttributes.ll [hunk 1]
```diff
@@ -35,9 +36,10 @@
 ; CHECK-SPIRV: 4 ExecutionMode [[FUNCENTRY]] 5894 1
 ; CHECK-SPIRV: 3 ExecutionMode [[FUNCENTRY]] 5895
 ; CHECK-SPIRV: 4 ExecutionMode [[FUNCENTRY]] 5896 8
+; CHECK-SPIRV: 4 ExecutionMode [[FUNCENTRY]] 5903 1000
 ; CHECK-SPIRV: 5 Function {{.*}} [[FUNCENTRY]] {{.*}}
 
-; CHECK-LLVM: define spir_kernel void {{.*}}kernel_name() {{.*}} !max_work_group_size ![[MAXWG:[0-9]+]] !no_global_work_offset ![[OFFSET:[0-9]+]] !max_global_work_dim ![[MAXWD:[0-9]+]] !num_simd_work_items ![[NUMSIMD:[0-9]+]]
+; CHECK-LLVM: define spir_kernel void {{.*}}kernel_name() {{.*}} !max_work_group_size ![[MAXWG:[0-9]+]] !no_global_work_offset ![[OFFSET:[0-9]+]] !max_global_work_dim ![[MAXWD:[0-9]+]] !num_simd_work_items ![[NUMSIMD:[0-9]+]] !scheduler_target_fmax_mhz ![[MAXMHZ:[0-9]+]]
```
Comment (mlychkov) [inline test/transcoding/IntelFPGAFunctionAttributes.ll:42]: Could you please add check for MAXMHZ value?

Comment (MrSidims) [reply test/transcoding/IntelFPGAFunctionAttributes.ll:42]: Done

**Review by mlychkov — APPROVED** [review]

**Comment by MrSidims** [comment]
@AlexeySotkin @AlexeySachkov  please take a look

File: lib/SPIRV/SPIRVInternal.h [hunk 2]
```diff
@@ -398,6 +398,7 @@ const static char MaxWGSize[] = "max_work_group_size";
 const static char NoGlobalOffset[] = "no_global_work_offset";
 const static char MaxWGDim[] = "max_global_work_dim";
 const static char NumSIMD[] = "num_simd_work_items";
+const static char FmaxMhz[] = "scheduler_target_fmax_mhz";
```
Comment (AlexeySotkin) [inline lib/SPIRV/SPIRVInternal.h:401]: A couple of minor comments/questions:
 - strictly speaking, not all of metadata names in this namespace are _SPIR_ metadata per https://www.khronos.org/registry/SPIR/specs/spir_spec-2.0.pdf. Should we move non-SPIR metadata names to a separate namespace?
 - should we have `intel` prefix or suffix for metadata introduced by Intel vendor extensions?

**Review by AlexeySotkin — APPROVED** [review]

File: lib/SPIRV/SPIRVInternal.h (hunk 2 above)
Comment (MrSidims) [reply lib/SPIRV/SPIRVInternal.h:401]: >> Should we move non-SPIR metadata names to a separate namespace?

That is a good suggestion. Let me do it in a separate patch.

>> should we have intel prefix or suffix for metadata introduced by Intel vendor extensions?

At first I was about to strongly agree with you (with a remark, that metadata names come from backends' developers, hence we need to align with them). Now I still agree with you, but also I think, that some of these names can become not only Intel-specific (in a certain circumstances). For example ivdep pragma/attribute is quiet popular in among compiler, see: https://gcc.gnu.org/onlinedocs/gcc-4.9.2/gcc/Loop-Specific-Pragmas.html .

**Comment by AlexeySotkin** [comment]
@MrSidims please resolve merge conflicts. Thanks.

**Comment by MrSidims** [comment]
> @MrSidims please resolve merge conflicts. Thanks.

Didn't have a suitable WS to build it fast. Let me check/debug through CI if I have resolved test conflict correctly.

**Comment by MrSidims** [comment]
@AlexeySotkin resolved, all good

**Pull request merged**